File: 최경열/이코테/구현/왕실의나이트/knight.js
const rl = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", (line) => {
  input.push(line);
}).on("close", () => {
  const row = parseInt(input[0][1]);
  const column = parseInt(input[0][0].charCodeAt(0) - "a".charCodeAt(0) + 1);
  let result = 0;
  const steps = [
    [-2, -1],
    [-2, 1],
    [2, -1],
    [2, 1],
    [-1, -2],
    [-1, 2],
    [1, -2],
    [1, 2],
  ];

  for (let i = 0; i < steps.length; i++) {
    let next_row = row + steps[i][0];
    let next_column = column + steps[i][1];
    if (
      next_row >= 1 &&
      next_row <= 8 &&
      next_column >= 1 &&
      next_column <= 8
    ) {
      result += 1;
    }
  }
  console.log(result);
});

File: 최경열/이코테/구현/왕실의나이트/knight.py
n =input()
row = int(n[1])
column = int(ord(n[0])) -int(ord('a'))+1
result=0
steps =[(-2,-1),(-2,1),(2,-1),(2,1),
        (-1,-2),(-1,2),(1,-2),(1,2)]

for i in steps:
    next_row = row+i[0]
    next_column=column+i[1]
    if next_row>=1 and next_row<=8 and next_column>=1 and next_column<=8:
        result+=1

print(result)

#해설
#1.미리 나이트가 이동할수 있는 경로(범위)를 배열로 지정
#2.이동경로(범위)를 순회하면서 나이트가 있는 x,y좌표를 가지고와
#  나이트있는자리에서 이동경로(범위)를 추가
#3.2번에 충족되고, 체스판사이즈(1~8)범위이내를 조건을 충적시키면 카운트증가
#4.전부 이동경로(범위)를 순회후 카운트값 출력